core/class/new_spec.rb

```diff
@@ -94,13 +94,11 @@ def message2; "hello"; end
     Class.new.superclass.should == Object
   end
 
-  it "raises a TypeError when given a non-Class" do
-    error_msg = /superclass must be a Class/
-    -> { Class.new("")         }.should raise_error(TypeError, error_msg)
-    -> { Class.new(1)          }.should raise_error(TypeError, error_msg)
-    -> { Class.new(:symbol)    }.should raise_error(TypeError, error_msg)
-    -> { Class.new(mock('o'))  }.should raise_error(TypeError, error_msg)
-    -> { Class.new(Module.new) }.should raise_error(TypeError, error_msg)
+  test_cases = ["", 1, :symbol, mock('o'), Module.new, BasicObject.new].each do |superclass|
+    it "raises a TypeError when given an instance of #{Kernel.instance_method(:class).bind(superclass).call.name}" do
```

Comment:
This executes a lot of code outside of `it`. Which means if an error happens for some Ruby implementation, it will fail to load the entire file and won't be able to run any of the specs in this file.
So, can you please keep it simple like it was and just add an extra line with `BasicObject.new`?

I can say from experience with ruby/spec that this trivial duplication is far less costly than all the downsides.

Reply:
I see, okay, makes sense. Just updated.